openedx/core/djangoapps/team_api/tests/test_views.py

```diff
@@ -0,0 +1,437 @@
+class TestTeamAPI(APITestCase, ModuleStoreTestCase):
+    """
+    Tests of the Team API
+    """
+
+    test_password = 'password'
+
+    topics_count = 10
+
+    def setUp(self):
+        super(TestTeamAPI, self).setUp()
+
+        teams_configuration = {
+            'topics':
+            [
+                {
+                    'id': 'topic_{}'.format(i),
+                    'team_count': i,
+                    'is_active': True,
+                    'name': 'Topic {}'.format(i),
+                    'description': 'Description for topic {}.'.format(i)
+                } for i in range(self.topics_count)
+            ] + [
+                {
+                    'id': 'inactive',
+                    'team_count': 0,
+                    'is_active': False,
+                    'name': 'Inactive Topic',
+                    'description': 'Inactive.'
+                }
+            ]
+        }
+
+        self.test_course_1 = CourseFactory.create(
+            org='TestX',
+            course='TS101',
+            display_name='Test Course',
+            teams_configuration=teams_configuration
+        )
+        self.test_course_2 = CourseFactory.create(org='MIT', course='6.002x', display_name='Circuits')
+
+        self.student_user = UserFactory.create(password=self.test_password)
+        self.student_user_not_active = UserFactory.create(password=self.test_password, is_active=False)
+        self.staff_user = UserFactory.create(password=self.test_password, is_staff=True)
+
+        self.test_team_1 = CourseTeamFactory.create(name='solar team', course_id=self.test_course_1.id, topic_id='renewable')
+        self.test_team_2 = CourseTeamFactory.create(name='Wind Team', course_id=self.test_course_2.id)
+        self.test_team_3 = CourseTeamFactory.create(name='Nuclear Team', course_id=self.test_course_1.id)
+        self.test_team_4 = CourseTeamFactory.create(name='coal team', course_id=self.test_course_2.id, is_active=False)
+
+        self.test_team_1.add_user(self.student_user)
+
+    def setup_inactive_user(self):
+        test_user = UserFactory.create(password=self.test_password)
+        self.client.login(username=test_user.username, password=self.test_password)
+        test_user.is_active = False
+        test_user.save()
+
+    def get_teams_list(self, expected_status, user=None, data=None):
+        user = user if user else self.student_user
+        self.client.login(username=user.username, password=self.test_password)
+        response = self.client.get(reverse('teams_list'), data=data)
+        self.assertEqual(expected_status, response.status_code)
+        return response
+
+    def get_teams_list_json(self, user=None, data=None):
+        response = self.get_teams_list(200, user=user, data=data)
+        return json.loads(response.content)
+
+    def test_list_teams_anonymous(self):
+        response = self.client.get(reverse('teams_list'))
+        self.assertEqual(403, response.status_code)
+
```

Comment:
We may also want a test that a user who has not verified their email address receives a 403.